File: app/src/main/java/sk/teamsoft/observableadapterdemo/DataView.java
package sk.teamsoft.observableadapterdemo;

import android.content.Context;
import android.support.annotation.AttrRes;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.util.AttributeSet;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.jakewharton.rxbinding2.view.RxView;

import java.lang.ref.WeakReference;

import io.reactivex.Observable;
import io.reactivex.functions.Function;
import sk.teamsoft.observablecollection.BindableView;

/**
 * @author Dusan Bartos
 *         Created on 15.04.2017.
 */

public class DataView extends LinearLayout implements BindableView<Data> {

    /**
     * No need for strong reference, because internal viewHolder implementation
     * keeps strong reference to shown item
     */
    private WeakReference<Data> itemReference;

    private TextView label;
    private TextView detail;

    public DataView(@NonNull Context context) {
        super(context);
    }

    public DataView(@NonNull Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    public DataView(@NonNull Context context, @Nullable AttributeSet attrs, @AttrRes int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override protected void onFinishInflate() {
        super.onFinishInflate();
        label = (TextView) findViewById(R.id.data);
        detail = (TextView) findViewById(R.id.data_detail);
    }

    @Override public void bindItem(Data item) {
        label.setText(item.label);
        detail.setText(item.detail);

        itemReference = new WeakReference<Data>(item);
    }

    @Override public Observable<Object> onObservableEvent() {
        return RxView.clicks(this)
                .map(new Function<Object, Object>() {
                    @Override
                    public Object apply(@io.reactivex.annotations.NonNull Object o) throws Exception {
                        return itemReference.get();
                    }
                });
    }
}

File: app/src/main/java/sk/teamsoft/observableadapterdemo/simple/DataView.kt
package sk.teamsoft.observableadapterdemo.simple

import android.content.Context
import android.util.AttributeSet
import android.widget.LinearLayout
import android.widget.TextView
import com.jakewharton.rxbinding2.view.RxView
import io.reactivex.Observable
import sk.teamsoft.observableadapterdemo.R
import sk.teamsoft.observablecollection.BindableView

/**
 * @author Dusan Bartos
 */
class DataView @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyleAttr: Int = 0
) : LinearLayout(context, attrs, defStyleAttr), BindableView<Data> {

    private val label: TextView by lazy { findViewById(R.id.data) }
    private val detail: TextView by lazy { findViewById(R.id.data_detail) }

    override val observableEvent: Observable<Any>?
        get() = RxView.clicks(this)

    override fun bindTo(item: Data) {
        label.text = item.label
        detail.text = item.detail
    }
}